### PR TITLE
Replaced which with command in shell function wrappers

### LINKED ISSUE #1496: Commands overridden by Docksal leads to infinite recursion on Fedora
**Description**
Similar to #1137, when installing via the `bash <(curl -fsSL https://get.docksal.io)` script, installation hangs here: 
```shell
Installing Docker...
# Executing docker install script, commit: 7cae5f8b0decc17d6571f9f52eb840fbc13b2737
+ sudo -E sh -c 'dnf install -y -q dnf-plugins-core'
+ sudo -E sh -c 'dnf config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo'
Adding repo from: https://download.docker.com/linux/fedora/docker-ce.repo
+ '[' stable '!=' stable ']'
+ sudo -E sh -c 'dnf makecache'
Docker CE Stable - x86_64                                                                                                                                                                                      89 kB/s | 3.5 kB     00:00    
Fedora 34 - x86_64                                                                                                                                                                                             60 kB/s |  12 kB     00:00    
Fedora 34 openh264 (From Cisco) - x86_64                                                                                                                                                                       18 kB/s | 989  B     00:00    
Fedora Modular 34 - x86_64                                                                                                                                                                                     32 kB/s |  12 kB     00:00    
Fedora 34 - x86_64 - Updates                                                                                                                                                                                  111 kB/s |  11 kB     00:00    
Fedora Modular 34 - x86_64 - Updates                                                                                                                                                                          101 kB/s |  11 kB     00:00    
google-chrome                                                                                                                                                                                                  17 kB/s | 1.3 kB     00:00    
Metadata cache created.
+ '[' -n '' ']'
+ sudo -E sh -c 'dnf install -y -q docker-ce'

Installed:
  containerd.io-1.4.4-3.1.fc34.x86_64  docker-ce-3:20.10.6-3.fc34.x86_64  docker-ce-cli-1:20.10.6-3.fc34.x86_64  docker-ce-rootless-extras-20.10.6-3.fc34.x86_64  docker-scan-plugin-0.7.0-3.fc34.x86_64  libcgroup-0.42.2-4.fc34.x86_64 

+ '[' -n 1 ']'
+ sudo -E sh -c 'dnf install -y -q docker-ce-rootless-extras'

================================================================================

To run Docker as a non-privileged user, consider setting up the
Docker daemon in rootless mode for your user:

    dockerd-rootless-setuptool.sh install

Visit https://docs.docker.com/go/rootless/ to learn about rootless mode.


To run the Docker daemon as a fully privileged service, but granting non-root
users access, refer to https://docs.docker.com/go/daemon-access/

WARNING: Access to the remote API on a privileged Docker daemon is equivalent
         to root access on the host. Refer to the 'Docker daemon attack surface'
         documentation for details: https://docs.docker.com/go/attack-surface/

================================================================================

groupadd: group 'docker' already exists
Client: Docker Engine - Community
 Version:           20.10.6
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        370c289
 Built:             Fri Apr  9 22:47:35 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8728dd2
  Built:            Fri Apr  9 22:45:20 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

**Steps to reproduce the issue:**

1. Install on Fedora 34

**Describe the results you received:**

See above.

**Describe the results you expected:**

Successful installation 

**Output of `fin config`:**

<details>
  <summary>fin config output</summary>

Like #1137, command hangs.

</details>

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

Like #1137, command hangs.

</details>

### PR DESCRIPTION
**Pull request by lmakarov**
- Replaced which with command in shell function wrappers: `pwd`, `docker`, `docker-compose` (see https://github.com/docksal/docksal/issues/1496#issuecomment-867231686)
- Dropped `docker-machine` wrapper (microsoft/WSL#4078 is now resolved)

Resolves #1496

### CODE REVIEW

File: bin/fin [hunk 1]
```diff
@@ -5054,7 +5041,7 @@ update ()
 	if is_linux; then
 		# Override docker to run it via sudo
 		docker () {
-			sudo "$(which docker)" "$@"
+			sudo command docker "$@"
```
Comment (lpeabody) [inline bin/fin:5044]: I don't think this works. I get this when I try this:

```
lpeabody@pop-os:~/Projects/eqrx-intranet (master)$ sudo command docker
sudo: command: command not found
```

**Review by lpeabody — CHANGES_REQUESTED** [review]

File: bin/fin (hunk 1 above)
Comment (lpeabody) [reply bin/fin:5044]: Googling, I found https://forums.opensuse.org/showthread.php/439150-sudo-lt-command-gt-command-not-found?s=62a70e125aab2af9c711eac9a69f792b&p=2167748#post2167748:

`sudo -i command docker` works for me, where as `sudo command docker` fails. I don't know if adding `-i` causes any other funny business...

Comment (lmakarov) [reply bin/fin:5044]: Interesting... What's OS/shell? Does this work? `command -v ls`

Comment (lpeabody) [reply bin/fin:5044]: Linux Pop!_OS 20.04
Linux pop-os 5.8.0-7642-generic #47~1614007149~20.04~82fb226-Ubuntu SMP Tue Feb 23 02:56:27 UTC  x86_64 x86_64 x86_64 GNU/Linux

```
lpeabody@pop-os:~/Projects/baxter-myclaria (feature/myclaria-site-profile)$ bash --version
bash --version
GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)
```

```
lpeabody@pop-os:~/Projects/docksal/service-cli (feature/composer-2-14)$ command ls
7.3  7.4  8.0  code-server  LICENSE  Makefile  README.md  tests
lpeabody@pop-os:~/Projects/docksal/service-cli (feature/composer-2-14)$ command -v ls
alias ls='ls --color=auto'
```

**Review by lpeabody — CHANGES_REQUESTED** [review]

File: bin/fin (hunk 1 above)
Comment (lmakarov) [reply bin/fin]: This works: `sudo $(command -v docker) "$@"`. Updated.

**Comment by lmakarov** [comment]
@lpeabody thanks for checking this!

While drilling into it, I identified a major bug in the `fin update` process - it used to exit with `0` even if the actual update failed. Addressed that as well as part of this PR (https://github.com/docksal/docksal/pull/1509/commits/3f6c4557fda5c765febf23e25693b160ae58d12e)

**Comment by lpeabody** [comment]
Happy to be of service! ✌️

**Review by lpeabody — APPROVED** [review]

**Pull request merged**